pypqlx.py

```diff
@@ -73,19 +73,35 @@ def __init__(self, first, last, ndata, N, S, C, L = ''):
         self.L = L if L != "" else "--"
     
     def mode(self):
-        pass
-    
+        mode = []
+        for i in sorted(set(pdf.period)):
+            a = pdf.power[max(pdf.count[pdf.period == i])]
+            mode.append(a)
+        return
+        
     def median(self):
         pass
     
     def average(self):
-        pass
+        avg = []
```

Comment:
all the other methods you used a full variable name, do the same here.

Reply:
Done!